dfdaemon/initializer/initializer.go

```diff
@@ -102,32 +101,37 @@ func cleanLocalRepo(options *options.Options) {
 }
 
 // rotateLog truncates the logs file by a certain amount bytes.
-func rotateLog(logFile *os.File, logFilePath string) {
+func rotateLog(logFile *os.File) {
 	logSizeLimit := int64(20 * 1024 * 1024)
 	for {
 		time.Sleep(time.Second * 60)
-		stat, err := os.Stat(logFilePath)
+		stat, err := logFile.Stat()
 		if err != nil {
-			log.Errorf("failed to stat %s: %s", logFilePath, err)
+			// make sure this error could be printed when logFile is broken
+			fmt.Fprintf(os.Stderr, "failed to stat: %s\n", err)
```

Comment:
I am not sure this error could be printed in stderr, since the fd stdout and stderr of process has been redirect to the logfile. Am I right? @godliness @mdianjun @lowzj @Starnop

Reply:
@allencloud 

Thanks, i think when we use `log.SetOutput(out io.Writer)` which means write log into the "out", the out could be a logfile and also could be std.out or std.err, here we write the log into std.err that i think it could be printed.